tfx/experimental/templates/taxi/e2e_tests/stub_kubeflow_e2e_test.py

```diff
@@ -0,0 +1,140 @@
+    logging.info('ENDPOINT: %s', self._endpoint)
+
+    self._base_container_image = '{}:{}'.format(self._BASE_CONTAINER_IMAGE,
+                                                random_id)
+    self._target_container_image = 'gcr.io/{}/{}:{}'.format(
+        self._GCP_PROJECT_ID, 'taxi-template-kubeflow-e2e-test', random_id)
+    self._record_dir = "gs://{}/testdata".format(self._BUCKET_NAME)
+    self._port_forwarding_process = self._setup_mlmd_port_forward()
+    self._prepare_base_container_image()
+    self._prepare_skaffold()
+
+  def tearDown(self):
+    super(StubKubeflowE2ETest, self).tearDown()
+    logging.info('Killing the GRPC port-forwarding process.')
+    self._port_forwarding_process.kill()
+    io_utils.delete_dir(self._record_dir)
+    self._cleanup_kfp()
+
+  def _cleanup_kfp(self):
+    self._cleanup_with_retry(self._delete_base_container_image)
+    self._cleanup_with_retry(self._delete_target_container_image)
+    self._cleanup_with_retry(self._delete_pipeline)
+    self._cleanup_with_retry(self._delete_pipeline_data)
+    self._cleanup_with_retry(self._delete_runs)
+
+  def testPipeline(self):
+    self._copy_template()
+    os.environ['KUBEFLOW_HOME'] = os.path.join(self._temp_dir, 'kubeflow')
+
+    # Uncomment all variables in config.
+    self._uncomment_multiline_variables(
+        os.path.join('pipeline', 'configs.py'), [
+            'GOOGLE_CLOUD_REGION',
+            'BIG_QUERY_WITH_DIRECT_RUNNER_BEAM_PIPELINE_ARGS',
+            'BIG_QUERY_QUERY', 'DATAFLOW_BEAM_PIPELINE_ARGS',
+            'GCP_AI_PLATFORM_TRAINING_ARGS', 'GCP_AI_PLATFORM_SERVING_ARGS'
+        ])
+
+    # Prepare data
+    self._prepare_data()
+    self._replace_file_content('kubeflow_dag_runner.py', [
+        ('DATA_PATH = \'gs://{}/tfx-template/data/\'.format(configs.GCS_BUCKET_NAME)',  # pylint: disable=line-too-long
+         'DATA_PATH = \'gs://{{}}/{}/{}\'.format(configs.GCS_BUCKET_NAME)'
+         .format(self._DATA_DIRECTORY_NAME, self._pipeline_name)),
+    ])
+
+    # Create a pipeline with all components.
+    updated_pipeline_file = self._add_all_components()
+    logging.info('Updated %s to add all components to the pipeline.',
+                 updated_pipeline_file)
+
+    self._create_pipeline()
+    self._run_pipeline()
+    self._check_telemetry_label()
+    logging.info("Successfully ran the pipeline.")
+    # Record pipeline outputs to self._record_dir
+    pipeline_recorder_utils.record_pipeline(
+        output_dir=self._record_dir,
+        metadata_db_uri=None,
+        host='localhost',
+        port=self._port,
+        pipeline_name=self._pipeline_name,
+        run_id=None)
+    self.assertTrue(tf.io.gfile.exists(self._record_dir))
+    logging.info("Pipeline has been recorded.")
+    # Enable stub executors
+    self._uncomment('kubeflow_dag_runner.py',
+                    ['supported_launcher_classes=[',
+                     '    stub_component_launcher.StubComponentLauncher',
+                     '],'])
```

Comment:
It looks like '],' is too broad. How about using '_uncomment_multiline_variables()' instead? It might require some changes in `_uncomment_multiline_variables()`.